shared/src/main/kotlin/com/egm/stellio/shared/model/CompactedEntity.kt

```diff
@@ -83,21 +136,30 @@ private fun filterMultiInstanceLanguageProperty(
     }
 
 private fun filterLanguageProperty(value: Map<String, Any>, transformationParameters: Map<String, String>?): Any {
-    val attributeCompactedType = AttributeCompactedType.forKey(value[JSONLD_TYPE_TERM] as String)!!
-    return if (attributeCompactedType == LANGUAGEPROPERTY) {
-        val localeRanges = Locale.LanguageRange.parse(transformationParameters?.get(QUERY_PARAM_LANG)!!)
-        val propertyLocales = (value[JSONLD_LANGUAGEMAP_TERM] as Map<String, Any>).keys.sorted()
-        val bestLocaleMatch = Locale.filterTags(localeRanges, propertyLocales)
-            .getOrElse(0) { _ ->
-                // as the list is sorted, @none is the first in the list if it exists
-                propertyLocales.first()
-            }
-        mapOf(
-            JSONLD_TYPE_TERM to NGSILD_PROPERTY_TERM,
-            JSONLD_VALUE_TERM to (value[JSONLD_LANGUAGEMAP_TERM] as Map<String, Any>)[bestLocaleMatch],
-            NGSILD_LANG_TERM to bestLocaleMatch
-        )
-    } else value
+    val languageFilter = transformationParameters?.get(QUERY_PARAM_LANG)!!
+    return value.let { entry ->
```

Comment:
why `value.let{}`?

Reply:
it is a way to apply a transformation on the map (`value`) as a whole